programs/sbf/tests/programs.rs

```diff
@@ -3956,7 +3956,7 @@ fn test_cpi_deprecated_loader_realloc() {
         if direct_mapping {
             assert_eq!(
                 result.unwrap_err().unwrap(),
-                TransactionError::InstructionError(0, InstructionError::InvalidRealloc)
+                TransactionError::InstructionError(0, InstructionError::ProgramFailedToComplete)
```

Comment:
@seanyoung I couldn't find any relevant log to still track the direct mapping related error in this case. Would you mind having a look?

Reply:
We used to expand memcpy in single instructions, so the error happened at a sBPF instruction. Now, we invoke the syscall, and the error happens in the syscall code. The error I got was `InvliadLength`.